standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RawStore.java

```diff
@@ -868,19 +895,6 @@ boolean getPartitionsByExpr(String catName, String dbName, String tblName,
   int getNumPartitionsByFilter(String catName, String dbName, String tblName, String filter)
     throws MetaException, NoSuchObjectException;
 
-  /**
-   * Get the number of partitions that match an already parsed expression.
-   * @param catName catalog name.
-   * @param dbName database name.
-   * @param tblName table name.
-   * @param expr an already parsed Hive expression
-   * @return number of matching partitions.
-   * @throws MetaException error accessing the RDBMS or working with the expression.
-   * @throws NoSuchObjectException no such table.
-   */
-  int getNumPartitionsByExpr(String catName, String dbName, String tblName, byte[] expr)
```

Comment:
Since this method is introduced in 3.0.0, can we mark this method as `Deprecated` instead of removing it, we can delete it in the next major release.

Reply:
Reverted.